soci/soci_index.go

```diff
@@ -305,8 +304,31 @@ func NewIndexBuilder(contentStore content.Store, blobStore orascontent.Storage,
 	}, nil
 }
 
-// Build builds a soci index for `img` and return the index with metadata.
+// Build builds a soci index for `img`, pushes it, and returns the index with metadata.
 func (b *IndexBuilder) Build(ctx context.Context, img images.Image) (*IndexWithMetadata, error) {
```

Comment:
Let's be explicit here where the index is pushed. i.e. ... for `img`, stores the index in the blob store, and returns ...

Also the index return value is no longer needed by the command line with this change. Should we go ahead and refactor to remove it?

Reply:
There's no other way to get the digest/info for a SOCI index if we don't return it here. I can imagine it might be useful for those using our library, and they can always choose to ignore it.